ext/spl/spl_array.c

```diff
@@ -460,6 +460,130 @@ static uint32_t spl_array_set_refcount(bool is_child, HashTable *ht, uint32_t re
 	return old_refcount;
 } /* }}} */
 
+/* For reasons ArrayObject must not go through the overloaded __set()/__get()/etc.
+ * magic methods, so to emit a warning for dynamic props (or Error on classes
+ * that do not support them, such as readonly classes) or an error for incompatible
+ * typed properties, we need to reimplement the logic of zend_std_write_property()
+ * somewhat. Thanks SPL. */
+static zval *zend_ignore_set_magic_method_write_property(
```

Comment:
`zend_std_write_property` skips magic methods if a property guard is set. Maybe you could use that to avoid reimplementing it. E.g. something like this:

```
uint32_t *guard = zend_get_property_guard(zobj, name);
uint32_t backup = *guard;
(*guard) |= IN_SET;
obj->handlers->write_property(...);
*guard = backup;
```

Reply:
Oh, I didn't know this, that might be very convenient